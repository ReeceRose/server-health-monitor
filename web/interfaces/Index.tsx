// Generic types

export interface StandardResponse {
  Data: unknown;
  StatusCode: number;
  Error: string;
  Success: boolean;
}

export interface Host {
  hostname?: string;
  uptime?: number;
  bootTime?: number;
  procs?: number;
  os?: string;
  platform?: string;
  platformFamily?: string;
  platformVersion?: string;
  kernelVersion?: string;
  kernelArch?: string;
  virtualizationSystem?: string;
  virtualizationRole?: string;
  hostID?: string;
  // TODO: add this to backend
  online?: boolean;
}

export interface Health {
  id: string;
  agentID: string;
  host: Host;
  createTime: number;
  updateTime: number;
  online?: boolean;
}

export interface HealthResponse extends StandardResponse {
  Data: Health[];
}
