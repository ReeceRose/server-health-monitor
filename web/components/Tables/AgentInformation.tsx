import { Host } from '../../interfaces/Index';

type Props = {
  online: boolean;
  host: Host;
};

const AgentInformation: React.FC<Props> = ({ online, host }) => {
  return (
    <div
      className={
        'relative flex flex-col break-words w-full mb-6 shadow-lg rounded bg-gray-700 text-white'
      }
    >
      <div className="px-4 py-3 mb-0 border-0 rounded-t">
        <div className="flex flex-wrap items-center">
          <div className="relative flex-1 flex-grow w-full max-w-full px-2">
            <h3 className={'font-semibold text-lg text-white'}>Servers</h3>
          </div>
        </div>
      </div>
      <div className="block w-full overflow-x-auto">
        <table className="items-center w-full bg-transparent border-collapse">
          <thead>
            <tr>
              <th
                className={
                  'px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500'
                }
              >
                Hostname
              </th>
              <th
                className={
                  'px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500'
                }
              >
                OS
              </th>
              <th
                className={
                  'px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500'
                }
              >
                Platform
              </th>
              <th
                className={
                  'px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500'
                }
              >
                Last Ping
              </th>
              <th
                className={
                  'px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500'
                }
              >
                Status
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th className="flex items-center p-4 px-6 text-xs text-left align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap">
                <span className="text-white">{host.hostname}</span>
              </th>
              <td className="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap">
                <span className="text-white">{host.os}</span>
              </td>
              <td className="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap">
                <span className="text-white">{host.platform}</span>
              </td>
              <td className="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap">
                TODO
              </td>
              <td className="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap">
                {online ? (
                  <span>
                    <i className="mr-2 text-green-400 fas fa-circle"></i> Online
                  </span>
                ) : (
                  <span>
                    <i className="mr-2 text-red-400 fas fa-circle"></i> Offline
                  </span>
                )}
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default AgentInformation;
