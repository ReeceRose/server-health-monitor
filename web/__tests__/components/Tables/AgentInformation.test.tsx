import { render } from '@testing-library/react';

import AgentInformation from '../../../components/Tables/AgentInformation';
import { Host } from '../../../interfaces/Index';

const host: Host = {
  os: 'ubunut',
  platform: 'linux',
  hostname: 'Test Machine',
};

test('AgentInformation component matches snapshot', () => {
  const { asFragment } = render(<AgentInformation host={host} online={true} />);
  expect(asFragment()).toMatchSnapshot();
});
