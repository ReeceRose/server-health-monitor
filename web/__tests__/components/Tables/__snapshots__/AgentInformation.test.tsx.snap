// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentInformation component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative flex flex-col break-words w-full mb-6 shadow-lg rounded bg-gray-700 text-white"
  >
    <div
      class="px-4 py-3 mb-0 border-0 rounded-t"
    >
      <div
        class="flex flex-wrap items-center"
      >
        <div
          class="relative flex-1 flex-grow w-full max-w-full px-2"
        >
          <h3
            class="font-semibold text-lg text-white"
          >
            Servers
          </h3>
        </div>
      </div>
    </div>
    <div
      class="block w-full overflow-x-auto"
    >
      <table
        class="items-center w-full bg-transparent border-collapse"
      >
        <thead>
          <tr>
            <th
              class="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500"
            >
              Hostname
            </th>
            <th
              class="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500"
            >
              OS
            </th>
            <th
              class="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500"
            >
              Platform
            </th>
            <th
              class="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500"
            >
              Last Ping
            </th>
            <th
              class="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500"
            >
              Status
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              class="flex items-center p-4 px-6 text-xs text-left align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap"
            >
              <span
                class="text-white"
              >
                Test Machine
              </span>
            </th>
            <td
              class="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap"
            >
              <span
                class="text-white"
              >
                ubunut
              </span>
            </td>
            <td
              class="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap"
            >
              <span
                class="text-white"
              >
                linux
              </span>
            </td>
            <td
              class="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap"
            >
              TODO
            </td>
            <td
              class="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap"
            >
              <span>
                <i
                  class="mr-2 text-green-400 fas fa-circle"
                />
                 Online
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
